src/index.ts

```diff
@@ -0,0 +1,56 @@
+import { characters } from './characters';
+import { clean, removeLNP } from './util';
+
+/** The current cache of all the supported alphabet characters  */
+export const alphabetMap: Record<string, string[]> = {};
+
+/** The current cache of all the supported confusable characters */
+export const confusablesMap: Record<string, string> = {};
+
+/** A cache of removed characters for optimal memory performance */
+const removeCache = new Map<string, string>();
+
+for (const [base, alts] of characters.entries()) {
+	alphabetMap[base] = [...alts];
+
+	for (const char of alts) {
+		confusablesMap[char] = base;
+	}
+}
+
+/**
+ * Removes confusable unicode characters from a string.
+ * @param str The text to remove confusables from.
+ */
+export function remove(str: string) {
+	if (removeCache.get(str)) return removeCache.get(str)!;
```

Comment:
For better performance, only perform a single access:
```suggestion
	const previous = removeCache.get(str);
	if (typeof previous === 'string') return previous;
```